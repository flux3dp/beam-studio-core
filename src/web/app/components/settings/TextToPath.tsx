/* eslint-disable @typescript-eslint/no-explicit-any */
import * as React from 'react';

import SelectControl from 'app/components/settings/SelectControl';
import useI18n from 'helpers/useI18n';

interface Props {
  fontSubstituteOptions: { value: any; label: string; selected: boolean }[];
  updateBeamboxPreferenceChange: (item_key: string, newVal: any) => void;
  defaultFontConvert: string;
}

function TextToPath({
  fontSubstituteOptions,
  updateBeamboxPreferenceChange,
  defaultFontConvert,
}: Props): JSX.Element {
  const lang = useI18n();
  const defaultLaserModuleOptions = React.useMemo(
    () => [
      { value: '1.0', label: '1.0', selected: defaultFontConvert === '1.0' },
      { value: '2.0', label: '2.0', selected: defaultFontConvert === '2.0' },
    ],
    [defaultFontConvert]
  );
  return (
    <>
      <div className="subtitle">{lang.settings.groups.text_to_path}</div>
      <SelectControl
        label={lang.settings.font_substitute}
        url={lang.settings.help_center_urls.font_substitute}
        id="font-substitue"
        options={fontSubstituteOptions}
        onChange={(e) => updateBeamboxPreferenceChange('font-substitute', e.target.value)}
      />
      <SelectControl
        label={lang.settings.font_convert}
        url={lang.settings.help_center_urls.font_convert}
        id="font-convert"
        options={defaultLaserModuleOptions}
        onChange={(e) => updateBeamboxPreferenceChange('font-convert', e.target.value)}
      />
    </>
  );
}

export default TextToPath;
