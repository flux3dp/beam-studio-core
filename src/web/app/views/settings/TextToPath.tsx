import * as React from 'react';

import i18n from 'helpers/i18n';
import SelectControl from 'app/views/settings/SelectControl';

interface Props {
  fontSubstituteOptions: { value: any, label: string, selected: boolean }[];
  updateBeamboxPreferenceChange: (item_key: string, newVal: any) => void;
}

function TextToPath({ fontSubstituteOptions, updateBeamboxPreferenceChange }: Props): JSX.Element {
  const lang = i18n.lang;
  return (
    <>
      <div className="subtitle">{lang.settings.groups.text_to_path}</div>
      <SelectControl
        label={lang.settings.font_substitute}
        url={lang.settings.help_center_urls.font_substitute}
        id="font-substitue"
        options={fontSubstituteOptions}
        onChange={(e) => updateBeamboxPreferenceChange('font-substitute', e.target.value)}
      />
    </>
  );
}

export default TextToPath;
