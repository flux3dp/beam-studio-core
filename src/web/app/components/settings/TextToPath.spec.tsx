import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('helpers/i18n', () => ({
  lang: {
    settings: {
      font_substitute: 'Substitute Unsupported Characters',
      help_center_urls: {
        font_substitute: 'https://support.flux3dp.com/hc/en-us/articles/360004496575',
      },
      groups: {
        text_to_path: 'Text',
      },
    },
  },
}));

// eslint-disable-next-line import/first
import TextToPath from './TextToPath';

test('should render correctly', () => {
  const updateBeamboxPreferenceChange = jest.fn();
  const wrapper = shallow(<TextToPath
    fontSubstituteOptions={[
      {
        value: 'TRUE',
        label: 'On',
        selected: true,
      },
      {
        value: 'FALSE',
        label: 'Off',
        selected: false,
      },
    ]}
    updateBeamboxPreferenceChange={updateBeamboxPreferenceChange}
  />);
  expect(toJson(wrapper)).toMatchSnapshot();

  wrapper.find('SelectControl').simulate('change', {
    target: {
      value: 'FALSE',
    },
  });
  expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(1);
  expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(1, 'font-substitute', 'FALSE');
});
