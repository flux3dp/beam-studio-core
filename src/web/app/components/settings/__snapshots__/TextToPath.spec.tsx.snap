// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Fragment>
  <div
    className="subtitle"
  >
    Text
  </div>
  <SelectControl
    id="font-substitue"
    label="Substitute Unsupported Characters"
    onChange={[Function]}
    options={
      [
        {
          "label": "On",
          "selected": true,
          "value": "TRUE",
        },
        {
          "label": "Off",
          "selected": false,
          "value": "FALSE",
        },
      ]
    }
    url="https://support.flux3dp.com/hc/en-us/articles/360004496575"
  />
</Fragment>
`;
